state/trackableDataTrie.go

```diff
@@ -48,7 +47,7 @@ func NewTrackableDataTrie(
 	}
 
 	return &trackableDataTrie{
-		tr:                  tr,
+		tr:                  nil,
```

Comment:
why removing trie from constructor? we'll rely on setting it later with SetDataTrie?

Reply:
When a new account is created, it does not have a data trie by default, so the trie will be nil at first. The data trie is created only when some data is added to the account.